src/EditorFeatures/CSharpTest/EditAndContinue/StatementEditingTests.cs

```diff
@@ -6933,6 +6943,182 @@ public void LocalFunction_RemoveAsync()
                 Diagnostic(RudeEditKind.ChangingFromAsynchronousToSynchronous, "local", FeaturesResources.local_function));
         }
 
+        [Fact]
+        public void LocalFunction_AddAttribute()
+        {
+            var src1 = "void L() { }";
+            var src2 = "[A]void L() { }";
+
+            var edits = GetMethodEdits(src1, src2);
```

Comment:
>GetMethodEdits [](start = 24, length = 14)

We should also check rude edits in these tests. That means we need to use GetTopEdits helper since the analyzer only operates on entire tree, not just method body. 
But should be easy enough to wrap each of the source snippets in `class C { void M() { ... } }`.